contracts/FlowCallbackScheduler.cdc

```diff
@@ -802,97 +793,168 @@ access(all) contract FlowCallbackScheduler {
-                            if callback.executionEffort <= lowPriorityEffortAvailable {
-                                lowPriorityEffortAvailable = lowPriorityEffortAvailable - callback.executionEffort
-                            } else {
-                                // remove the callback from this timestamp
-                                // and try to add it to a future timestamp
-                                let newTimestamp = self.calculateScheduledTimestamp(
-                                    timestamp: timestamp + 1.0, 
-                                    priority: Priority.Low, 
-                                    executionEffort: callback.executionEffort
-                                )!
-
-                                let callbackResource <- self.garbageCollect(callback: callback)
-                                callbackResource.setScheduledTimestamp(newTimestamp: newTimestamp)
-                                self.addCallback(slot: newTimestamp, callback: <-callbackResource)
-                                
-                                continue
+                
+                    switch callback.priority {
+                        case Priority.High:
+                            high.append(callback)
+                        case Priority.Medium:
+                            medium.append(callback)
+                        case Priority.Low:
+                            // add low priority callbacks only if there is space left in the available effort
+                            if callback.executionEffort <= lowEffortLeft {
+                                low.append(callback)
                             }
-                        }
-
-                        emit PendingExecution(
-                                id: id,
-                                priority: callback.priority.rawValue,
-                                executionEffort: callback.executionEffort,
-                                fees: callback.fees.balance,
-                                callbackOwner: callback.handler.address
-                            )
-
-                        // after pending execution event is emitted we set the callback as executed because we 
-                        // must rely on execution node to actually execute it. Execution of the callback which is 
-                        // done in a separate transaction that calls executeCallback(id) function can not update 
-                        // the status of callback or any other shared state, since that blocks concurrent callback 
-                        // execution. Therefore optimistic update to executed is made here to avoid race condition.
-                        callback.setStatus(newStatus: Status.Executed)
-
-                        // charge the fee for callback execution
-                        destroy callback.payAndRefundFees(refundMultiplier: 0.0)
+                            lowEffortLeft = lowEffortLeft - callback.executionEffort
```

Comment:
Shouldn't the subtraction be within the if statement? We only want to subtract it from the effort left if we have added it to the pending callbacks list

Reply:
yes